providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/FlagStore.java

```diff
@@ -0,0 +1,101 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process.storage;
+
+import dev.openfeature.contrib.providers.flagd.resolver.process.model.FeatureFlag;
+import dev.openfeature.contrib.providers.flagd.resolver.process.model.FlagParser;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.Connector;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.StreamPayload;
+import lombok.extern.java.Log;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+
+/**
+ * Feature flag storage.
+ */
+@Log
+public class FlagStore implements Storage {
+    private final Object sync = new Object();
+    private final AtomicBoolean shutdown = new AtomicBoolean(false);
+    private final BlockingQueue<StorageState> stateBlockingQueue = new LinkedBlockingQueue<>(1);
+    private final Map<String, FeatureFlag> flags = new ConcurrentHashMap<>();
+
+    private final Connector connector;
+
+    public FlagStore(final Connector connector) {
+        this.connector = connector;
+    }
+
+    /**
+     * Initialize storage layer.
+     */
+    public void init() {
+        connector.init();
+        Thread streamer = new Thread(() -> {
+            try {
+                streamerListener(connector);
+            } catch (InterruptedException e) {
+                log.log(Level.WARNING, "connection listener failed", e);
+            }
+        });
+        streamer.setDaemon(true);
+        streamer.start();
+    }
+
+    /**
+     * Shutdown storage layer.
+     */
+    public void shutdown() {
+        shutdown.set(true);
```

Comment:
since we have the info, shall we protect the method from being called multiple times?
```suggestion
     if(!shutdown.getAndSet(true)){
```

Reply:
Good one, will update

Reply:
Done :)